docs/source/ttnn/ttnn/adding_new_ttnn_operation.rst

```diff
@@ -7,174 +7,98 @@ Adding New ttnn Operation
-
-    Tensor <new_operation>(const Tensor &input_tensor, bool some_arg);
 
-    }  // namespace tt_metal
-    }  // namespace tt
+What is a ttnn operation?
+-------------------------
 
+A ttnn operation is a function that takes in one or more input tensors and produces one or more output tensors. It is implemented in C++ and can be called from Python.
 
-.. note:
+What steps are needed to add ttnn operation in C++?
+---------------------------------------------------
+1. (Optional) Implement device operation in C++. Device operation is a struct that specifies how to create output tensors and a program to run on the device. If the ttnn operation is composed of other ttnn operations, then you can skip this step.
+2. Implement ttnn operation in C++ and register it using `ttnn::register_operation`.
 
-    If you need optional input tensors or would like to pass in optional output tensors, then refer to :doc:`Operations </ttnn/dependencies/tt_lib>` for how to write ops that use them
+What steps are needed to add ttnn operation in Python?
+------------------------------------------------------
+1. Take an existing registerd C++ operation and add a Python binding for it using `ttnn::bind_registered_operation`.
+2. In python, decorate the operation using `ttnn.register_operation`. (This step will be deprecated in the future)
 
 
-Add `tt_eager/tt_dnn/op_library/<new_operation>/<new_operation>.cpp`:
 
-.. code-block:: cpp
-
-    #include "tt_metal/host_api.hpp"
-    #include "tt_dnn/op_library/run_operation.hpp"
-
-    namespace tt {
-    namespace tt_metal {
-
-
-    void <NewOperation>::validate(const std::vector<Tensor> &input_tensors) const {
-        ...
-    }
-
-    std::vector<Shape> <NewOperation>::compute_output_shapes(const std::vector<Tensor> &input_tensors) const {
-        std::vector<Shape> output_shapes = ...;
-        return output_shapes;
-    }
-
-    std::vector<Tensor> create_output_tensors(const std::vector<Tensor> &input_tensors) const {
-        std::vector<Tensor> output_tensors = ...;
-        return output_tensors;
-    }
-
-    operation::ProgramWithCallbacks create_program(const std::vector<Tensor>& input_tensors, std::vector<Tensor> &output_tensors) const {
-        Program program = ...;
-        return operation::ProgramWithCallbacks{program};
-    }
-
-    };
+C++ Implementation
+------------------
 
-    Tensor <new_operation>(const Tensor &input_tensor, bool some_arg) {
-        std::vector<Tensor> input_tensors = {input_tensor};
-        std::vector<Tensor> output_tensors operation::run(DeviceOperation(<NewOperation>{some_arg}, {input_tensor}));
-        return output_tensors[0];
-    }
+Step 1: Implement device operation (Optional)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-    }  // namespace tt_metal
-    }  // namespace tt
+In order to add a new device operation, follow the directory structure shown below:
 
+`ttnn/cpp/ttnn/operations/<category>/<operation_name>/device/<operation_name>_device_operation.hpp`
```

Comment:
I am a little concerned about category part. People often mess up with categorizaton, specially when it is not 100% clear.. I am ok to keep it like that, just want to raise a Q for discussion.